gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/RepositoryExplorer.java

```diff
@@ -48,44 +48,59 @@ public class RepositoryExplorer {
     private static final Pattern UNSTABLE_VERSION_PATTERN = Pattern.compile(
             ".*(-rc(-?\\d+)?|-SNAPSHOT|-M\\d+|-alpha(-?\\d+)?|-beta(-?\\d+)?)$", Pattern.CASE_INSENSITIVE);
 
-    private final Cache<CacheKey, Set<GroupPartOrPackageName>> folderCache = Caffeine.newBuilder()
+    private final Cache<String, Set<GroupPartOrPackageName>> folderCache = Caffeine.newBuilder()
             .expireAfterWrite(10, TimeUnit.MINUTES)
             .maximumSize(100)
             .build();
 
+    // In general, we don't want to be caching version data as it changes often. However, for wildcard complete it
+    // can be very expensive to repeatedly get data that realistically doesn't change on a second by second basis so
+    // having a short-lived cache is okay
+    private final Cache<String, Set<DependencyVersion>> shortLivedVersionCache = Caffeine.newBuilder()
+            .expireAfterWrite(2, TimeUnit.MINUTES)
+            .maximumSize(10000)
+            .build();
+
     public final Set<GroupPartOrPackageName> getGroupPartOrPackageName(DependencyGroup group, String url) {
-        CacheKey cacheKey = CacheKey.of(url, group);
-        Set<GroupPartOrPackageName> folders = folderCache.get(cacheKey, key -> {
```

Comment:
This could result in returning null if load folders was cancelled half way through